modelzoo/MMoE/train.py

```diff
@@ -0,0 +1,673 @@
+        if intra:
+            sess_config.intra_op_parallelism_threads = intra
+        hooks = []
+
+        hooks.append(tf.train.StopAtStepHook(last_step=train_steps))
+        hooks.append(tf.train.LoggingTensorHook(
+                        {
+                            'steps': model.global_step,
+                            'loss': model.loss
+                        },
+                     every_n_iter=100))
+
+        scaffold = tf.train.Scaffold(
+            local_init_op=tf.group(tf.global_variables_initializer(),
+                                   tf.local_variables_initializer(), train_init_op))
+
+        with tf.train.MonitoredTrainingSession(
+                master=server.target,
+                is_chief=tf_config['is_chief'],
+                checkpoint_dir=checkpoint_dir,
+                scaffold=scaffold,
+                hooks=hooks,
+                log_step_count_steps=100,
+                config=sess_config) as sess:
+            while not sess.should_stop():
+                _, train_loss = sess.run([model.train_op, model.loss])
+    else:
+        with tf.Session(config=sess_config) as sess:
+            sess.run(tf.global_variables_initializer())
+            sess.run(tf.local_variables_initializer())
+            merged = tf.summary.merge_all()
+            writer = tf.summary.FileWriter(checkpoint_dir, sess.graph)
+            saver = tf.train.Saver(tf.global_variables(),
+                                max_to_keep=keep_checkpoint_max)
+            options = tf.RunOptions(trace_level=tf.RunOptions.FULL_TRACE)
+            run_metadata = tf.RunMetadata()
+
+            # train model
+            sess.run(train_init_op)
+
+            start = time.perf_counter()
+            for _in in range(0, train_steps):
+                if args.save_steps > 0 and (_in % args.save_steps == 0
+                                            or _in == train_steps - 1):
+                    _, train_loss, events = sess.run([model.train_op, model.loss, merged])
+                    writer.add_summary(events, _in)
+                    checkpoint_path = saver.save(sess,
+                                                save_path=os.path.join(
+                                                checkpoint_dir,
+                                                'MMOE_checkpoint'),
+                                                global_step=_in)
+                    print(f'Save checkpoint to {checkpoint_path}')
+                elif (args.timeline > 0 and _in % args.timeline == 0):
+                    _, train_loss = sess.run([model.train_op, model.loss],
+                                            options=options,
+                                            run_metadata=run_metadata)
+                    fetched_timeline = timeline.Timeline(run_metadata.step_stats)
+                    chrome_trace = fetched_timeline.generate_chrome_trace_format()
+                    print(f'Save timeline to {checkpoint_path}')
```

Comment:
“checkpoint_path” should be "checkpoint_drive".  （This comment is proposed by Wang Duyi）

Reply:
Changed `checkpoint_path` to `checkpoint_dir`